download_and_create_reference_datasets/v02/mito/create_ht__gnomad_38.py

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env python3
+
+# Download gnomAD mito datset vcf file, import it to a matrix table, annotate required fields into al hail table,
+# and save it to a file.
+
+import logging
+import hail as hl
+
+logging.basicConfig(format='%(asctime)s %(levelname)-8s %(message)s')
+logger = logging.getLogger()
+
+CONFIG = {
+    'vcf_path': 'gs://gcp-public-data--gnomad/release/3.1/vcf/genomes/gnomad.genomes.v3.1.sites.chrM.vcf.bgz',
```

Comment:
you can downlaod the current reference data table from here if you want to play around with it and see what we have for mito: `gs://seqr-reference-data/GRCh38/all_reference_data/combined_reference_data_grch38.ht`

Reply:
The current reference data is within the v2 subfolder which is not intuitive at all `gs://seqr-reference-data/GRCh38/all_reference_data/v2/combined_reference_data_grch38-2.0.4.ht`. This  will be part of my seqr storage restructuring/policy project because nothing makes sense in there.

Reply:
I've just checked these alternatives of the data source. I found the gnomad genomes build 38 doesn't include mito data. But there is a hail table available for the gnomAD mito-DNA (gs://gcp-public-data--gnomad/release/3.1/ht/genomes/gnomad.genomes.v3.1.sites.chrM.ht). We could use this one instead of the VCF. I checked the existing reference data (both gs://seqr-reference-data/GRCh38/all_reference_data/combined_reference_data_grch38.ht and gs://seqr-reference-data/GRCh38/all_reference_data/v2/combined_reference_data_grch38-2.0.4.ht). They don't include all the fields we need for mito. It seems that we have to create another combined reference dataset for mito.

Reply:
I know they don't have ALL the fields we need for mito. But do they have ANY of the fields we need for mito? If so, we should get whatever fields they do have from the existing reference data. For example, for gnomad, what information is in the existing reference data, and what fields are missing? Again, that reference table is made off of literally the same download you are using so I am not going to approve a code change that requires us to download part of a file we've literally already downloaded

Reply:
The AF_hom, AC_hom, AF_het, AC_het, and max_hl fields are missing. Only the AN is not missing. But, I found all gnomad ANs are NA.

Reply:
@mike-w-wilson I'll defer to you on this one - we are going to be making a mito-specific reference data table (its going to include some mito only data sources like helix), but for the annotations that are shared between the main SNP pipeline and the mito piepeline we'd like to use the main reference data file so we aren't duplicating data and efforts.  We need some additional fields from gnomad that are currently not being pulled from the download when building the main reference table. We can either 

1.  Change the script that writes our main reference table from gnomad to pull the additional annotations we need
2. Do a separate download of the gnomad mito variants and add the additional annotations to the mito-only reference table

Option 1 means we would only be keeping one copy of gnomad data, which seems good, but option 2 means we wouldn;t have any potentially unused nnotations on the main reference data, which also seems good. So your call which you think is the better approach

Reply:
So I think I am missing something. The gnomad release HT  does not contain any mito calls, its just autosomes and sex chromosomes, mito is filtered out before the release, so the current reference data does not have any mito calls. Mito calls are in `gs://gcp-public-data--gnomad/release/3.1/ht/genomes/gnomad.genomes.v3.1.sites.chrM.ht`. What fields are missing from the chrM HT? Or are you going to use the `custom_gnomad_select_v3` https://github.com/broadinstitute/hail-elasticsearch-pipelines/blob/f92dbf7886ef55cef381ea5241d105c99678d1f3/download_and_create_reference_datasets/v02/hail_scripts/write_combined_reference_data_ht.py#L209 for mito and would like to adjust it?  Either way, I don't think this script is necessary as you're just selecting out the fields you want and rewriting it to storage we have to pay for. I would just pull the public HT and select the fields you want, which is what the combined ref script does, when you are running all the other annotations like helix which I'm guessing are coming in a different script?  Again, sorry if I am missing something. 

I think I like the idea of one large reference but agree maintenance could be cumbersome and I'm not sure about creating a bunch of NA entries across the entire combined reference for all of the mito annotations...so I guess I'm leaning toward 2 if I have understood the objective correctly.

Reply:
@mike-w-wilson You got most of the information. And you are right. If you like to know more you can refer to the [document](https://docs.google.com/document/d/1AeBJIhzMWlhBhuJJOnqrUMHcAkRQlSub1kKrq_JLRfM/edit#) about what the fields we need and the data sources for them.

Reply:
Ah, I misunderstood I thought ` gs://gcp-public-data--gnomad/release/3.1/ht/genomes/gnomad.genomes.v3.1.sites.chrM.ht` was just a subset of `gs://gcp-public-data--gnomad/release/3.1/ht/genomes/gnomad.genomes.v3.1.sites.ht`, not a different file entirely. Sorry for adding confusion